cmake/math/MathLibs.cmake

```diff
@@ -145,19 +145,8 @@ else()
     set(_blacs_lib)
 endif()
 
-# miro: for MKL 10.0.1.014
-set(MKL_BLAS_LIBS ${_scalapack_lib} ${_compiler_mkl_interface}${_lib_suffix} ${_thread_lib} mkl_core mkl_def mkl_mc ${_blacs_lib} guide pthread m)
-#  try this MKL BLAS combination with SGI MPT
-set(MKL_BLAS_LIBS2 ${_scalapack_lib} ${_compiler_mkl_interface}${_lib_suffix} ${_thread_lib} mkl_core ${_blacs_lib}   guide pthread m)
-# newer MKL BLAS versions do not have libguide
-set(MKL_BLAS_LIBS3 ${_scalapack_lib} ${_compiler_mkl_interface}${_lib_suffix} ${_thread_lib} mkl_core ${_blacs_lib}         pthread m)
-# ancient MKL BLAS
-set(MKL_BLAS_LIBS4 mkl guide m)
-
-set(MKL_LAPACK_LIBS mkl_lapack95${_lib_suffix} ${_compiler_mkl_interface}${_lib_suffix})
-
-# older MKL LAPACK
-set(MKL_LAPACK_LIBS2 mkl_lapack)
+set(MKL_BLAS_LIBS mkl_rt pthread m dl)
+set(MKL_LAPACK_LIBS mkl_rt pthread m dl)
```

Comment:
(1) Why the `dl`?
(2) Why repeat the sealer libs for blas and lapack when they're returned as lapackblas anyway?
(3) Why kill off the ability to return a non-mkl_rt library list by removing all the MKL_BLAS_LIBS{N} signatures? Why not just insert your new one as MKL_BLAS_LIBS and increment all the others by one?
(4) Would be nice to be able to bypass the mkl_rt option with a cmakevar.